solvers/bcd_celer.py

```diff
@@ -0,0 +1,363 @@
+from benchopt import BaseSolver, safe_import_context
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from numpy.linalg import norm
+    from numba import njit
+    from mtl_utils.common import sum_squared, _get_dgemm, norm_l21
+
+
+if import_ctx.failed_import:
+
+    def njit(f):  # noqa: F811
+        return f
+
+
+def dual_mtl(alpha, norm_Y2, Theta, Y):
+    """
+    Problem solved:
+    min 0.5 * (lambda ** 2) * ||(Y / lambda) - Theta|| ** 2 + 0.5 * ||Y|| ** 2
+    """
+    d_obj = - ((Y / alpha - Theta) ** 2).sum()
+    d_obj *= 0.5 * alpha ** 2
+    d_obj += norm_Y2 / 2.
+    return d_obj
+
+
+def primal_mtl(W, alpha, R):
+    """
+    Problem solved:
+    min 0.5 * ||Y - XB|| ** 2 + lambda * ||B||_{2, 1}
+    """
+    p_obj = sum_squared(R) / 2
+    p_obj += norm_l21(W, 1, copy=False) * alpha
+    return p_obj
+
+
+@njit
+def set_prios_mtl(X, W, norms_X_block, prios, screened, radius,
+                  n_screened, norm_XT_Theta):
+    n_features = X.shape[1]
+    n_tasks = W.shape[1]
+    for j in range(n_features):
+        if screened[j] or norms_X_block[j] == 0:
+            prios[j] = np.inf
+            continue
+        nrm = norm_XT_Theta[j]
+        prios[j] = (1. - nrm) / norms_X_block[j]
+        if prios[j] > radius:
+            for k in range(n_tasks):
+                if W[j, k] != 0:
+                    break
+            else:
+                screened[j] = True
+                n_screened += 1
+    return n_screened
+
+
+@njit
+def dual_scaling_mtl(Theta, X, C, skip):
+    n_features = X.shape[1]
+    norm_XT_Theta = np.zeros(n_features)
+    nrm = 0
```

Comment:
suggestion : simplify the code by removing this and in the last line: return np.max(norm_XT_Theta)